components/admin_console/system_user_detail/system_user_detail.tsx

```diff
@@ -33,26 +38,44 @@ import SheidOutlineIcon from 'components/widgets/icons/shield_outline_icon.jsx';
 
 import './system_user_detail.scss';
 
-export default class SystemUserDetail extends React.PureComponent {
-    static propTypes = {
-        user: PropTypes.object.isRequired,
-        mfaEnabled: PropTypes.bool.isRequired,
-        isDisabled: PropTypes.bool,
-        actions: PropTypes.shape({
-            updateUserActive: PropTypes.func.isRequired,
-            setNavigationBlocked: PropTypes.func.isRequired,
-            addUserToTeam: PropTypes.func.isRequired,
-        }).isRequired,
+export type Props = {
+    user: UserProfile;
+    mfaEnabled: boolean;
+    isDisabled?: boolean;
+    actions: {
+        updateUserActive: (userId: string, active: boolean) => Promise<ActionResult | ActionResult[]>;
+        setNavigationBlocked: (blocked: boolean) => void;
+        addUserToTeam: (teamId: string, userId?: string) => Promise<{data: TeamMembership; error?: any}>;
     }
+}
 
-    static defaultProps = {
+export type State = {
+    teams: Record<string, any> | null;
+    teamIds: Array<string> | null;
```

Comment:
Maybe we can be more expressive here and use `Array<$ID<Team>>`

Reply:
Updated.